lib/apiservers/engine/backends/container.go

```diff
@@ -828,11 +828,11 @@ func (c *Container) ContainerRm(name string, config *types.ContainerRmConfig) er
 		}
 	}
 
-	if err := c.containerProxy.Remove(vc, config); err != nil {
-		return err
+	operation := func() error {
+		return c.containerProxy.Remove(vc, config)
 	}
 
-	return nil
+	return retry.Do(operation, IsConflictError)
```

Comment:
I believe the only time we get this is when Remove from the portlayer returns a conflict error. There is the soft power off race condition, though the conflict error could also be completely legitimate. I think we should change from using a default `BackoffConfig` here. as this could cause a `docker rm` to hang for upwards of 30 seconds when we are getting a legitimate rm against an online container.

Reply:
I would add a 20 second max to avoid weird Host Sync issues.(not sure what happened to my second comment... it seems to have randomly disappeared)

Reply:
max interval of 1 second possibly as well.

Reply:
It's a legitimate performance concern. I have an update that should address that.